app/controllers/discovered_hosts_controller.rb

```diff
@@ -24,11 +25,46 @@ def index
   end
 
   def show
-    # filter graph time range
     @range = nil
-
-    # summary report text
     @report_summary = nil
+    regex_hightlights = Setting[:discovery_CATEGORY_highlights].empty? ? /^(productname|memorysize|interface|manufacturer|architecture|macaddress$|processorcount|physicalprocessorcount|discovery_boot|ipaddress$)/ : Regexp.new(eval(Setting[:discovery_CATEGORY_highlights]))
+    regex_storage = Setting[:discovery_CATEGORY_storage].empty? ? /^blockdevice/ : Regexp.new(eval(Setting[:discovery_CATEGORY_storage]))
+    regex_hardware = Setting[:discovery_CATEGORY_hardware].empty? ? /^(hardw|manufacturer|memo|process)/ : Regexp.new(eval(Setting[:discovery_CATEGORY_hardware]))
+    regex_network = Setting[:discovery_CATEGORY_network].empty? ? /^(ipaddress|interfaces|dhcp|fqdn|hostname|link|mtu|net)|(#{@host.facts_hash["interfaces"].gsub(",","|")})/ : Regexp.new(eval(Setting[:discovery_CATEGORY_network]))
+    regex_software = Setting[:discovery_CATEGORY_software].empty? ? /^(bios|os|discovery)/ : Regexp.new(eval(Setting[:discovery_CATEGORY_software]))
+    regex_ipmi = Setting[:discovery_CATEGORY_ipmi].empty? ? /^ipmi/ : Regexp.new(eval(Setting[:discovery_CATEGORY_ipmi]))
+
+    @facts = @host.facts_hash.clone
+    @facts.delete("memorysize_mb")
+    @general = {}
+    @categories = {}
+    @categories[:Storage] = {}
+    @categories[:Network] = {}
+    @categories[:Hardware] = {}
+    @categories[:Software] = {}
+    @categories[:IPMI] = {}
+    @categories[:Miscellaneous] = {}
+    @facts.each do |key, value|
+      value = number_to_human_size(value)  if /size$/.match(key)
+      case key
+        when regex_hightlights
+          @general[key] = value
+        when regex_storage
+          @categories[:Storage][key] = value
+        when regex_hardware
+          @categories[:Hardware][key] = value
+        when regex_software
+          @categories[:Software][key] = value
+        when regex_network
+          @categories[:Network][key] = value
+        when regex_ipmi
+          @categories[:IPMI][key] = value
+        else
+          @categories[:Miscellaneous][key] = value
+      end
+      @facts.delete(key)
+    end
+    @categories.each {|key,val| @categories.delete(key) if @categories[key].empty? }
```

Comment:
Why don't you put regexps into an array, so you can iterate over groups. Then you can get rid of lines 41-46 and also get rid of that case statement (iterate through all regexps, if any match then determine group name, put it into the correct group name). Then it will be much shorter.